apps/condo/domains/contact/components/EditContactForm.tsx

```diff
@@ -0,0 +1,224 @@
+import { useOrganization } from '@core/next/organization'
+import get from 'lodash/get'
+import { useRouter } from 'next/router'
+import { Contact } from '../utils/clientSchema'
+import LoadingOrErrorPage from '@condo/domains/common/components/containers/LoadingOrErrorPage'
+import { Loader } from '@condo/domains/common/components/Loader'
+import { FormWithAction } from '@condo/domains/common/components/containers/FormList'
+import { UserAvatar } from '@condo/domains/user/components/UserAvatar'
+import { Rule } from 'rc-field-form/lib/interface'
+import { PhoneInput } from '@condo/domains/common/components/PhoneInput'
+import { FormResetButton } from '../../common/components/FormResetButton'
+import { Button } from '../../common/components/Button'
+import { canManageContacts } from '../../organization/permissions'
+const { normalizePhone } = require('@condo/domains/common/utils/phone')
+
+const INPUT_LAYOUT_PROPS = {
+    labelCol: {
+        span: 11,
+    },
+    wrapperCol: {
+        span: 13,
+    },
+    style: {
+        paddingBottom: '24px',
+        maxWidth: '453px',
+    },
+}
+
+export const EditContactForm: React.FC = () => {
+    const intl = useIntl()
+    const ErrorMessage = intl.formatMessage({ id: 'errors.PdfGenerationError' })
+    const LoadingMessage = intl.formatMessage({ id: 'Loading' })
+    const ContactNotFoundTitle = intl.formatMessage({ id: 'Contact.NotFound.Title' })
+    const ContactNotFoundMessage = intl.formatMessage({ id: 'Contact.NotFound.Message' })
+    const ProfileUpdateTitle = intl.formatMessage({ id: 'EditingContact' })
+    const NameLabel = intl.formatMessage({ id: 'field.FullName.short' })
+    const FullNameRequiredMessage = intl.formatMessage({ id: 'field.FullName.requiredError' })
+    const FieldIsRequiredMessage = intl.formatMessage({ id: 'FieldIsRequired' })
+    const EmailErrorMessage = intl.formatMessage({ id: 'pages.auth.EmailIsNotValid' })
+    const PhoneIsNotValidMessage = intl.formatMessage({ id: 'pages.auth.PhoneIsNotValid' })
+    const FullNamePlaceholderMessage = intl.formatMessage({ id:'field.FullName' })
+    const PhoneLabel = intl.formatMessage({ id: 'Phone' })
+    const ExamplePhoneMessage = intl.formatMessage({ id: 'example.Phone' })
+    const EmailLabel = intl.formatMessage({ id: 'field.EMail' })
+    const ApplyChangesMessage = intl.formatMessage({ id: 'ApplyChanges' })
+    const NoPermissionMessage = intl.formatMessage({ id: 'EditingContactNoPermission' })
+
+    const { query, push } = useRouter()
+    const { organization, link } = useOrganization()
+    const contactId = get(query, 'id', '')
+    const {
+        obj: contact,
+        loading,
+        error,
+        refetch,
+    } = Contact.useObject({
+        where: {
+            id: String(contactId),
+            organization: {
+                id: String(organization.id),
+            },
+        },
+    })
+
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
```

Comment:
Is there way to do this action without typescript warning?

Reply:
Yep...removed it ;)

Reply:
@sitozzz I've done a research and here's what I found:
Typescript want us to pass IContactFormState...which contains required fields such as "name" and "phone"
But we don't know their values here...since it will be obtained from form.
There're few ways to fix it:
1. Change Interface (which can lead to global changes in other places since we'll need 2 Interfaces for this (with required name and phone for form and non-required for hooks)
2. Pass default values in {} (This way will not give us right to overwrite it later :(    )
3. Left "// @ts-ignore" here

So, to say shortly...we cannot remove this comment yet